### PR TITLE
Minor corrections to documentation

### PR DESCRIPTION
**Pull request by davidtheclark**
While working on a Flow type interface for Studio I noticed a few more little typos and oversights to fix.

### CODE REVIEW

File: js/util/browser/browser.js [hunk 1]
```diff
@@ -66,14 +66,6 @@ exports.timed = function (fn, dur, ctx) {
...
 };
 
-/**
- * Test if the current browser supports Mapbox GL JS
- * @param {Object} options
- * @param {boolean} [options.failIfMajorPerformanceCaveat=false] Return `false`
- *   if the performance of Mapbox GL JS would be dramatically worse than
- *   expected (i.e. a software renderer would be used)
- * @return {boolean}
- */
```
Comment (lucaswoj) [inline js/util/browser/browser.js]: Why did you delete this?

Comment (lucaswoj) [inline js/util/browser/browser.js]: Ah. Duplicate docs. 👍

File: js/ui/map.js [hunk 2]
```diff
@@ -1236,6 +1236,11 @@ function removeNode(node) {
   * Returns a Boolean indicating whether the browser [supports Mapbox GL JS](https://www.mapbox.com/help/mapbox-browser-support/#mapbox-gl-js).
   *
   * @function supported
+  * @param {Object} options
+  * @param {boolean} [options.failIfMajorPerformanceCaveat=false] If `true`,
+  *   the function will return `false` if it is determined
+  *   that the performance of the created WebGL context would be too low.
```
Comment (lucaswoj) [inline js/ui/map.js]: Can we use the copy from the deleted documentation instead? It is the product of additional rounds of revision. 

```diff
/**
  * Test if the current browser supports Mapbox GL JS
  * @param {Object} options
  * @param {boolean} [options.failIfMajorPerformanceCaveat=false] Return `false`
  *   if the performance of Mapbox GL JS would be dramatically worse than
  *   expected (i.e. a software renderer would be used)
  * @return {boolean}
  */
```

Comment (davidtheclark) [inline js/ui/map.js]: Yes. I will replace that sentence in both places it appears.

File: js/ui/map.js [hunk 3]
```diff
@@ -417,8 +417,8 @@ util.extend(Map.prototype, /** @lends Map.prototype */{
      * If the map's current zoom level is higher than the new maximum,
      * the map will zoom to the new maximum.
      *
-     * @param {number} maxZoom The maximum zoom level to set (0-20).
-     *   If`null` or `undefined` is provided, the function removes the current maximum zoom (sets it to 20).
+     * @param {number | null | undefined} maxZoom The maximum zoom level to set (0-20).
```
Comment (lucaswoj) [inline js/ui/map.js]: Why not `?number`?

Comment (davidtheclark) [inline js/ui/map.js]: That would be fine, too. I just copied from above for consistency. I will change both.

File: js/ui/camera.js [hunk 4]
```diff
@@ -513,7 +515,7 @@ util.extend(Camera.prototype, /** @lends Map.prototype */{
      * @param {number} [options.screenSpeed] The average speed of the animation measured in screenfuls
      *     per second, assuming a linear timing curve. If `options.speed` is specified, this option is ignored.
      * @param {Function} [options.easing] An easing function for the animated transition.
-     * @param {EventData} [eventData] Data to propagate to any event listeners.
+     * @param {Object} [eventData] Data to propagate to any event listeners.
```
Comment (lucaswoj) [inline js/ui/camera.js]: Can we be more specific with these type signatures?

Comment (davidtheclark) [inline js/ui/camera.js]: `EventData` had been rendered meaningless, right? I had thought from prior conversation that this data you could send in is an arbitrary object. Is that wrong?

File: js/ui/control/geolocate.js [hunk 5]
```diff
@@ -94,6 +94,6 @@ Geolocate.prototype = util.inherit(Control, {
  * @event error
  * @memberof Geolocate
  * @instance
- * @property {EventData} data The returned PositionError object from the callback in [Geolocation.getCurrentPosition()](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition)
+ * @property {PositionError} data The returned [PositionError](https://developer.mozilla.org/en-US/docs/Web/API/PositionError) object from the callback in [Geolocation.getCurrentPosition()](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition).
```
Comment (davidtheclark) [inline js/ui/control/geolocate.js]: @lucaswoj For context: I added these links because the documentation generator does not understand those particular data types.

File: js/ui/camera.js (hunk 4 above)
Comment (lucaswoj) [inline js/ui/camera.js]: You're right. This type signature is correct 👍

**Comment by lucaswoj** [comment]
Looks good! 🚢
